rust/ql/lib/codeql/rust/dataflow/internal/DataFlowImpl.qll

```diff
@@ -839,14 +839,28 @@ module RustDataFlow implements InputSig<Location> {
     SsaFlow::localMustFlowStep(_, node1, node2)
   }
 
-  class LambdaCallKind = Void;
+  class LambdaCallKind = Unit;
 
-  // class LambdaCallKind;
   /** Holds if `creation` is an expression that creates a lambda of kind `kind` for `c`. */
-  predicate lambdaCreation(Node creation, LambdaCallKind kind, DataFlowCallable c) { none() }
+  predicate lambdaCreation(Node creation, LambdaCallKind kind, DataFlowCallable c) {
+    exists(Expr cl | cl = creation.asExpr().getExpr() and cl = c.asCfgScope() |
+      cl instanceof ClosureExpr or cl instanceof AsyncBlockExpr
```

Comment:
I don't think we should include `AsyncBlockExpr` here, otherwise I guess we would want to have different `LambdaCallKind` for closures and async blocks?

Reply:
I think it makes sense to treat them as lambdas (they capture variables, are "called" by `.await`). But you're right that we should also have a lambda kind for them. That's what I was alluding to in the PR description, that the async block handling is half baked. I can take `AsyncBlockExpr` for now and leave it until we implement proper support?